packages/nextjs/lib/rules/pages-need-get-link-function-export.ts

```diff
@@ -152,7 +135,7 @@ export default creator<Options, MessageIds>({
                     return
                 }
 
-                const [_, ...pagePathSegments] = path.split("pages")
+                const [_, ...pagePathSegments] = filename.split("pages")
```

Comment:
We should remove the actual full base path instead of splitting at a string that could appear in any name